pytorch_blade/pytorch_blade/torch-mlir/lib/Conversion/TorchToMhlo/DiscTorchToMhlo.cpp

```diff
@@ -169,6 +169,42 @@ class ConvertAtenUnaryOp : public OpConversionPattern<AtenOpT> {
 };
 } // namespace
 
+namespace {
+template <typename AtenOpT>
+class ConvertAtenExtractOp : public OpConversionPattern<AtenOpT> {
+ public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+  LogicalResult matchAndRewrite(
+      AtenOpT op,
+      OpAdaptor adaptor,
+      ConversionPatternRewriter& rewriter) const override {
+    auto input = adaptor.getOperands()[0];
+    auto inpTy = input.getType().template dyn_cast<RankedTensorType>();
+    if (!inpTy)
+      return op.emitError("only RankedTensorType is supported");
+
+    auto outTy = OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
+        op.getType());
+    auto elemTy = inpTy.getElementType();
+
+    if (outTy != elemTy) {
+      auto output =
+          rewriter.create<tensor::ExtractOp>(op.getLoc(), elemTy, input);
```

Comment:
A little bit confused ...  `tensor::ExtractOp` is usually created given a tensor `Value` and an indices `Value`s, that does the first argument `elemTy` mean here?

Reply:
`elemTy` is given as the new output type. The `indices` parameter is optional.